app/ui-utils/client/lib/RoomHistoryManager.js

```diff
@@ -234,7 +232,15 @@ export const RoomHistoryManager = new (class extends Emitter {
 
 		waitAfterFlush(() => {
 			const heightDiff = wrapper.scrollHeight - previousHeight;
-			wrapper.scrollTop = scroll + heightDiff;
+			let toScroll = wrapper.scrollHeight;
+			let len = Math.ceil(Math.log10(wrapper.scrollHeight + 1));
+
+			while (len > 2 && heightDiff < toScroll) {
+				toScroll -= heightDiff;
+				len = Math.ceil(Math.log10(toScroll + 1));
+			}
+
+			wrapper.scrollTop = toScroll + heightDiff;
```

Comment:
Ok, so, like the clever algorithm ;) 

But, two points, 1) code's glancability is too low; 2) don't think it still solves the problem;

Can you check a suspicion of mine please? There, if i'm not  mistaken, is a *race condition*. Think about someone else in the same room sending messages while another person is  attempting to load history. Multiple people at the same time doing the same. You save the *current* (really the key word should be "*instanteneous*") height, do your calculations, but in the meantime the div height is still increasing/has increased. That invalidates any calculations done on the old data. And we're left with random scrolls.

Reply:
Thanks for the review, yeah right, my bad, I missed `race condition`, to re-verify the code for this specific case while loading history I sent a too-long message, and yeah it broke. Well, I did verify this scenario for the current in-use code and it also breaks so should I revert the changes to `scrollIntoView` it scrolls the DOM thrice but works fine in this scenario. Please let me know whether or not we're going with this (scrollIntoView).
Thank you!

Reply:
> it scrolls the DOM thrice but works fine in this scenario  

Definitely like scrollIntoView, but don't scroll three times. u can just clean up the other scrolling bit. 

Before implementing can you also check if there's any cross-browser/cross-version compatibility issues with this method?

Another thing imo would be great is if you can highlight the last seen message after scrolling to the right place for  like a second. IIRC this same file has example for this - uses jquery. Because of the inevitable two scroll situation and crowdedness, even if the view positions itself correctly reliably, it kinda becomes hard to track the last viewed message. The highlight will be able to help with that.

Additionally, please go through the code multiple times, and really clean up any redundant code. Thanks!

Reply:
Thank you for the valuable feedback debdutdeb and Sing-Li , I'll try to clean up as much as possible and will also add a highlight on the last message. 
I'll be looking forward to your guidance if got stuck anywhere. Thank you!

Reply:
I have removed the thrice scrolling part and also added the highlight function to the last seen message; here is a vid showing the changes.

https://user-images.githubusercontent.com/61188295/156938325-a69a118f-e8f0-4673-aebc-ead8638ef07e.mp4

Please have a review. Thank you!